src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -424,6 +436,22 @@ private boolean addProvisionedSlave(@Nonnull PodTemplate template, @CheckForNull
         PodList namedList = client.pods().inNamespace(templateNamespace).withLabels(labelsMap).list();
         List<Pod> namedListItems = namedList.getItems();
 
+        if (this.isCapOnlyOnUnDeadPods()) {
```

Comment:
instead of getting the full list, then filtering here, can't it be done using something like `withFields("status.phase", "running|pending")` not sure what would be the right syntax but should be possible based on https://github.com/kubernetes/kubernetes/issues/49387

Reply:
I can do 
```
./kubectl get pods --field-selector=status.phase='Succeeded'
```
I can also do:
```
./kubectl get pods --field-selector=status.phase='Failed'
```
But I can't do:
```
./kubectl get pods --field-selector=status.phase='Succeeded|Failed'
```
Neither can I do:
```
./kubectl get pods --field-selector='status.phase in (Succeeded, Failed)'
```

I couldn't find any official documentation on fieldselector :cry: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#

I also looked at the implementation of [withFields](https://github.com/fabric8io/kubernetes-client/blob/v3.1.0/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Filterable.java#L42) and [getFieldQueryParam](https://github.com/fabric8io/kubernetes-client/blob/v3.1.0/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java#L576-L588) and it doesn't seem like a selector has been implemented yet. :sob:

Reply:
However, I can do:
```
./kubectl get pods --field-selector='status.phase!=Failed,status.phase!=Succeeded,status.phase!=Unknown'
```
But again, we really need a function `withFieldSelector()` at https://github.com/fabric8io/kubernetes-client/blob/v3.1.0/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Filterable.java